Python/bytecodes.c

```diff
@@ -5316,7 +5322,6 @@ dummy_func(
         }
 
         tier2 op(_START_EXECUTOR, (executor/4 --)) {
```

Comment:
Side note, this instruction now exists solely to set the current executor in the tier two interpreter. We should try to remove it as a follow-up (doesn't need to be in this PR).

Reply:
We can replace all uses of `current_executor` with an operand, which will remove the `current_executor` local variable from the tier 2 interpreter and the `_JIT_EXECUTOR` patch from the JIT template.

We can also shrink `_CHECK_VALIDITY` a bit by replacing `current_executor->vm_data.valid` with `*validity_ptr`:
```C
tier2 op(_CHECK_VALIDITY, (validity_ptr/4 --)) {
    DEOPT_IF(*validity_ptr == 0);
}
```

For another PR.